src/NATS.Client.Core/NatsOpts.cs

```diff
@@ -31,9 +31,9 @@ public sealed record NatsOpts
 
     public ILoggerFactory LoggerFactory { get; init; } = NullLoggerFactory.Instance;
 
-    public int WriterBufferSize { get; init; } = 1048576;
+    public int WriterBufferSize { get; init; } = 65536;
 
-    public int ReaderBufferSize { get; init; } = 1048576;
+    public int ReaderBufferSize { get; init; } = 65536;
```

Comment:
A couple comments on ReaderBufferSize:

1. Does reducing this value have a noticeable effect on reading large messages, such as the Perf test with 1MiB message size?

2. Looks like it is [used in NatsKVStore.cs](https://github.com/nats-io/nats.net.v2/blob/5e66e0bd2daccdca1a129b74983a1baddb13c2fd/src/NATS.Client.KeyValueStore/NatsKVStore.cs#L270) - will decreasing it have an effect?  Or is its usage there incorrect?

Reply:
> 1. Does reducing this value have a noticeable effect on reading large messages, such as the Perf test with 1MiB message size?

Haven't tried that but even if it is making a difference on loopback (I'm guessing being that large would be less effective on e.g. ethernet). also my impression is large messages are exception and discouraged so application developer should increase the buffer it they want.

> 2. Looks like it is [used in NatsKVStore.cs](https://github.com/nats-io/nats.net.v2/blob/5e66e0bd2daccdca1a129b74983a1baddb13c2fd/src/NATS.Client.KeyValueStore/NatsKVStore.cs#L270) - will decreasing it have an effect?  Or is its usage there incorrect?

Good spot! That looks like a bug. just created #385

Reply:
pub sub with 1mb payload

# Buffer Size = 64K
![image](https://github.com/nats-io/nats.net.v2/assets/386903/caaf6f37-075a-41b6-86de-57dfe06b9fdf)
![image](https://github.com/nats-io/nats.net.v2/assets/386903/625ee780-0f85-4529-b3ca-397bbdae32ef)
```
ReaderBufferSize:65,536
WriterBufferSize:65,536
# Sub synced
snapshot_10
snapshot_50
snapshot_90
pub time: 00:00:19.4860152, sync: 1, async: 9999
sub time: 00:00:19.4920023
Memory: 39.75 MB
1,026 msgs/sec ~ 1,026.05 MB/sec
```

# Buffer Size = 1M
![image](https://github.com/nats-io/nats.net.v2/assets/386903/ede50fa0-d056-4716-9c34-1beef72d6402)
![image](https://github.com/nats-io/nats.net.v2/assets/386903/677e9bda-a444-43ac-924c-5b8bb93bf9fe)

```
ReaderBufferSize:1,048,576
WriterBufferSize:1,048,576
# Sub synced
snapshot_10
snapshot_50
snapshot_90
pub time: 00:00:19.0241384, sync: 3, async: 9997
sub time: 00:00:19.0306999
Memory: 42.29 MB
1,051 msgs/sec ~ 1,050.93 MB/sec
```

# Program.cs
```csharp
using System.Buffers;
using System.Diagnostics;
using System.Threading.Channels;
using JetBrains.Profiler.Api;
using NATS.Client.Core;

MemoryProfiler.CollectAllocations(true);

//const int size = 1048576;
const int size = 65536;

const int max = 10_000;
const int payloadSize = 1048576;

var natsOpts = new NatsOpts
{
    Url = "127.0.0.1:4333",
    ReaderBufferSize = size,
    WriterBufferSize = size,
    SubPendingChannelFullMode = BoundedChannelFullMode.Wait,
};

Console.WriteLine($"ReaderBufferSize:{natsOpts.ReaderBufferSize:n0}");
Console.WriteLine($"WriterBufferSize:{natsOpts.WriterBufferSize:n0}");

await using var nats1 = new NatsConnection(natsOpts);
await using var nats2 = new NatsConnection(natsOpts);

var cts = new CancellationTokenSource(TimeSpan.FromSeconds(60));

await nats1.PingAsync();
await nats2.PingAsync();

var subActive = 0;
var subReader = Task.Run(async () =>
{
    var count = 0;
    await using var sub = await nats1.SubscribeCoreAsync<NatsMemoryOwner<byte>>("x", cancellationToken: cts.Token);
    await foreach (var msg in sub.Msgs.ReadAllAsync(cts.Token))
    {
        using (msg.Data)
        {
            if (msg.Data.Length == 1)
            {
                Interlocked.Increment(ref subActive);
                continue;
            }

            if (++count == max)
            {
                break;
            }

            if (count == (max / 10) * 1)
            {
                Console.WriteLine("snapshot_10");
                MemoryProfiler.GetSnapshot("snapshot_10");
            }

            if (count == (max / 10) * 5)
            {
                Console.WriteLine("snapshot_50");
                MemoryProfiler.GetSnapshot("snapshot_50");
            }

            if (count == (max / 10) * 9)
            {
                Console.WriteLine("snapshot_90");
                MemoryProfiler.GetSnapshot("snapshot_90");
            }
        }
    }
});

// Ensure subscription is active
while (Interlocked.CompareExchange(ref subActive, 0, 0) == 0)
{
    await Task.Delay(1000);
    await nats2.PublishAsync("x", 1, cancellationToken: cts.Token);
}

Console.WriteLine("# Sub synced");

var stopwatch = Stopwatch.StartNew();

var payload = new ReadOnlySequence<byte>(new byte[payloadSize]);
var pubSync = 0;
var pubAsync = 0;
for (var i = 0; i < max; i++)
{
    var vt = nats2.PublishAsync("x", payload, cancellationToken: cts.Token);
    if (vt.IsCompletedSuccessfully)
    {
        pubSync++;
        vt.GetAwaiter().GetResult();
    }
    else
    {
        pubAsync++;
        await vt;
    }
}

Console.WriteLine("pub time: {0}, sync: {1}, async: {2}", stopwatch.Elapsed, pubSync, pubAsync);
await subReader;
Console.WriteLine("sub time: {0}", stopwatch.Elapsed);

var seconds = stopwatch.Elapsed.TotalSeconds;

var meg = Math.Pow(2, 20);

var totalMsgs = 2.0 * max / seconds;
var totalSizeMb = 2.0 * max * payloadSize / meg / seconds;

var memoryMb = Process.GetCurrentProcess().PrivateMemorySize64 / meg;

Console.WriteLine("Memory: {0:n2} MB", memoryMb);
Console.WriteLine($"{totalMsgs:n0} msgs/sec ~ {totalSizeMb:n2} MB/sec");

```